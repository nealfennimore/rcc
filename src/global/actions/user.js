import { createAction } from 'redux-actions';

export const SET_USERNAME = 'SET_USERNAME';
export const setUsername = createAction( SET_USERNAME, ( username ) => ( { username } ) );

export const SET_TOKEN = 'SET_TOKEN';
export const setToken = createAction( SET_TOKEN, ( token ) => ( { token } ) );
