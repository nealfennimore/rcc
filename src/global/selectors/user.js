import { createSelector } from 'reselect';

const getSlice = ( state ) => state.user;

export const getUsername = createSelector( getSlice, ( user ) => user.username );
export const getToken = createSelector( getSlice, ( user ) => user.token );
