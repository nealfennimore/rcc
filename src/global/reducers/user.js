import { SET_USERNAME, SET_TOKEN } from 'global/actions/user';

const INITIAL_STATE = {
    username: null,
    token: null,
};

export default function user( state = INITIAL_STATE, action ) {
    switch ( action.type ) {
    case SET_USERNAME:
    case SET_TOKEN:
        return {
            ...state,
            ...action.payload,
        };
    default:
        return state;
    }
}
