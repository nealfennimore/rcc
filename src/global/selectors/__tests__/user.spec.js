import React from 'react';
import renderer from 'react-test-renderer';
import * as user from '../user';

describe( 'User Selectors', () => {
    const state = {
        user: {
            token: 'token',
            username: 'username',
        },
    };

    Object.entries( user ).forEach( ( [selectorName, selector] ) => {
        it( selectorName, () => {
            expect( selector( state ) ).toMatchSnapshot();
        } );
    } );
} );
