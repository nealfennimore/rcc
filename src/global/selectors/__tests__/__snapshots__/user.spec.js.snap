// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Selectors getToken 1`] = `"token"`;

exports[`User Selectors getUsername 1`] = `"username"`;
